src/register/RegistrationFailure.jsx

```diff
@@ -1,47 +1,55 @@
 import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 import { Alert } from '@edx/paragon';
-
-const hasNoErrors = (userErrors) => (
-  userErrors.every((errorList) => (!errorList[0]))
-);
+import { INTERNAL_SERVER_ERROR } from '../login/data/constants';
+import messages from './messages';
 
 const RegistrationFailureMessage = (props) => {
   const errorMessage = props.errors;
+  const { errorCode } = props.errors;
   const userErrors = [];
 
   useEffect(() => {
     window.scrollTo({ left: 0, top: 0, behavior: 'smooth' });
   }, [props.submitCount]);
 
-  Object.keys(errorMessage).forEach((key) => {
-    const errors = errorMessage[key];
-    const errorList = errors.map((error) => (
-      (error.user_message) ? (
-        <li key={error} className="text-left">
-          {error.user_message}
+  let serverError;
+  switch (errorCode) {
+    case INTERNAL_SERVER_ERROR:
+      serverError = (
+        <li key={INTERNAL_SERVER_ERROR} className="text-left">
+          {props.intl.formatMessage(messages['registration.request.server.error'])}
         </li>
-      ) : null
-    ));
-    userErrors.push(errorList);
-  });
+      );
+     userErrors.push(serverError);
+     break;
+
+     default:
+        Object.keys(errorMessage).forEach((key) => {
+          const errors = errorMessage[key];
+          const errorList = errors.map((error) => (
+            (error.user_message) ? (
+              <li key={error} className="text-left">
+                {error.user_message}
+              </li>
+            ) : null
+          ));
+          userErrors.push(errorList);
```

Comment:
Why we are pushing an array into another array here? I guess we should `userErrors.concat(errorList);` instead of push.

Reply:
@waheedahmed  `userErrors` must contain a list of `li` elements in order to render them on the top of the page.Concat doesn't work here because it will concat all the li's in each successive iteration.